packages/@aws-cdk/aws-rds/lib/engine.ts

```diff
@@ -0,0 +1,53 @@
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { compare } from './private/version';
+import { ParameterGroupFamily } from './props';
+
+/** @internal */
+export abstract class BaseEngine {
+  /**
+   * The engine.
+   */
+  public readonly name: string;
+
+  /**
+   * The single user secret rotation application.
+   */
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
```

Comment:
Yesterday I learnt - we should be about classes marked @internal, especially when they have public APIs. https://github.com/aws/jsii/issues/1734 Be careful here.

They don't fail on tests in the package since they directly use the `.ts`, and not the generated `.d.ts` file.

Check that the public getters and methods are, in fact, available and work when used when imported from outside this package.

Reply:
Hmm, very interesting. You're correct - I've just tried it from a different module, and it indeed fails. This is interesting, but very concerning at the same time... module-private abstract base classes are very commonly used in the CDK (things like `BucketBase`, etc.), so I wonder whether this is some bug in JSII...

For now, I've moved everything to one file, and that seems to have helped. We can split into 2 files again when we introduce interfaces like `IClusterEngine` instead of classes.

Reply:
Also, I just noticed that the PR build failed on the `@aws-cdk/aws-backup` module, which depends on RDS:

```
@aws-cdk/aws-backup: [96m../aws-rds/lib/cluster-engine.d.ts[0m:[93m1[0m:[93m10[0m - [91merror[0m[90m TS2305: [0mModule '"./engine"' has no exported member 'BaseEngine'.
@aws-cdk/aws-backup: [7m1[0m import { BaseEngine } from './engine';
@aws-cdk/aws-backup: [7m [0m [91m         ~~~~~~~~~~[0m
@aws-cdk/aws-backup: [96m../aws-rds/lib/instance-engine.d.ts[0m:[93m1[0m:[93m10[0m - [91merror[0m[90m TS2305: [0mModule '"./engine"' has no exported member 'BaseEngine'.
@aws-cdk/aws-backup: [7m1[0m import { BaseEngine } from './engine';
@aws-cdk/aws-backup: [7m [0m [91m         ~~~~~~~~~~[0m
@aws-cdk/aws-backup: Error: /codebuild/output/src026615207/src/github.com/aws/aws-cdk/node_modules/jsii/bin/jsii --project-references --silence-warnings=reserved-word exited with error code 1
@aws-cdk/aws-backup: Build failed. ${timers.display()}
```

So luckily our tests caught the issue as well.

Reply:
> So luckily our tests caught the issue as well.

Yep, but only if a library downstream uses that code path. Not commonly the case.